<a href="https://colab.research.google.com/github/JipFlinterman/TuringMachineJip/blob/main/PROJECTTMDL-Week1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# The titanic project


We started our project by finding a dataset. We thought of the titanic dataset on Kaggle. We thought it would be interesting to see **what kind of people will survive based on different charasteristics**. We have got a training set, and a test set. We will use the training set for learning the "machine" what type of people will survive. After that we will use this program to predict if the people in the test set will survive the titanic.

In [29]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

#LOADING THE DATASETS
df_train = pd.read_csv("/content/train.csv")
df_test = pd.read_csv("/content/test.csv")

**DATA EXPLANATION:**

---


survival - Survival - [0 = No, 1 = Yes]
pclass - Ticket class - [1 = 1st, 2 = 2nd, 3 = 3rd]
sex	- Sex
Age	- Age in years
sibsp	- # of siblings / spouses aboard the Titanic
parch	- # of parents / children aboard the Titanic
ticket - Ticket number
fare - Passenger fare
cabin	- Cabin number
embarked	- Port of Embarkation	- [C = Cherbourg, Q = Queenstown, S = Southampton]


# Making an overview of the data

In [11]:
#PRINTING AN OVERVIEW OF WHICH COLUMNS ARE IN THE DATASET
print(df_train.columns)
print(df_test.columns)

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')
Index(['PassengerId', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch',
       'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')


In [49]:
#LOOKING FOR WEIRD THINGS LIKE MISSING VALUES
print(df_train.describe())
print(df_test.describe())
print("-----------------------------------")
print(df_train.head())

       PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200  
       PassengerId      Pclass         Age       SibSp       Parch        Fare
co

# Filtering out data
We decided to drop the rows with missing ages. This made our dataset go down from 891 to 712 observations. We also decided to drop the cabin column since it has a lot of missing values. We don't want to work with incomplete information and therefore we dropped it. We also decided to make sex binary and give males the value 0 and females the value 1.



In [50]:
#DROPPING CABIN
df_train_filtered = df_train.drop('Cabin', axis=1)

#REMOVING MISSING VALUES
df_train_filtered = df_train_filtered.dropna()

#MAKING SEX BINARY
df_train_filtered["Sex"] = [0 if x == 'male' else 1 for x in df_train_filtered["Sex"]]

#SMALL CHECK
print(df_train_filtered.head(10))

    PassengerId  Survived  Pclass  \
0             1         0       3   
1             2         1       1   
2             3         1       3   
3             4         1       1   
4             5         0       3   
6             7         0       1   
7             8         0       3   
8             9         1       3   
9            10         1       2   
10           11         1       3   

                                                 Name  Sex   Age  SibSp  \
0                             Braund, Mr. Owen Harris    0  22.0      1   
1   Cumings, Mrs. John Bradley (Florence Briggs Th...    1  38.0      1   
2                              Heikkinen, Miss. Laina    1  26.0      0   
3        Futrelle, Mrs. Jacques Heath (Lily May Peel)    1  35.0      1   
4                            Allen, Mr. William Henry    0  35.0      0   
6                             McCarthy, Mr. Timothy J    0  54.0      0   
7                      Palsson, Master. Gosta Leonard    0   2.0   